src/tools/id-mapping/types/idMappingSearchResults.ts

```diff
@@ -1,13 +1,25 @@
 /* Results as given by the server */
+import { APIModel } from '../../../shared/types/apiModel';
 import { UniParcAPIModel } from '../../../uniparc/adapters/uniParcConverter';
 import { UniProtkbAPIModel } from '../../../uniprotkb/adapters/uniProtkbConverter';
 import { UniRefAPIModel } from '../../../uniref/adapters/uniRefConverter';
 
-export type Mapping = {
+export type MappingAPIModel = {
   from: string;
   to: string | UniProtkbAPIModel | UniRefAPIModel | UniParcAPIModel;
 };
 
 export type IDMappingSearchResults = {
-  results: Mapping[];
+  results: MappingAPIModel[];
 };
+
+export type MappingTo = {
+  to: string;
+};
+
+export type MappingFrom = {
+  from: string;
+};
+
+// Should this be a subset of APIMode, and if so how do we handle it in usePagination?
+export type MappingFlat = (APIModel | MappingTo) & MappingFrom;
```

Comment:
Should this be a subset of APIMode, and if so how do we handle it in usePagination?

Reply:
If you want usePagination to work with IDMapping search results you should add those definitions to APIModel directly where it's defined I think. The from in MappingFrom should then be optional, and maybe you could add it only to the 4 namespaces supported for ID Mapping.